### DISCUSSION TITLE
Best practices for securely receiving events

### DISCUSSION THREAD

**Discussion by caseyjlaw** [chime-frb-open-data/community#82] [Q&A] [answered]
There is great science potential in using FRB alerts from CHIME and elsewhere. However, automatic responses to alerts are also risky, since unauthorized users may get access to observatory control systems.

I am in a discussion with staff at OVRO about how to safely enable alert-driven observations (esp from CHIME/FRB). All OVRO observatory control software is set up behind a gateway server. A request to open a port is scrutinized and we need to demonstrate that software that listens on an open port is secure.

How are people in the CHIME/FRB community building software to receive VOEvents securely? Can you share any tips for tools or configuration that help us let events in and keep the bad guys out?

**Comment by tabbott36** [comment, marked as answer]
Hi Casey, 

Thanks for bringing this up and apologies for the delayed response. Assuming you are using [Comet](https://comet.transientskp.org/en/stable/installation.html) to receive events, Section 6 of [Swinbank's Comet Broker Paper](https://doi.org/10.1016/j.ascom.2014.09.001) may be of interest. It describes this issue with respect to the broader VOEvent community and proposes two possible solutions. Notably for Comet users, there may be an experimental version of the Comet broker that make use of OpenPGP (a software that supports authentication of VOEvents through cryptographic signatures). It is supposedly available upon request, however, it has not been implemented for CHIME/FRB VOEvents. It may be worth looking into to avoid issues like the one you're experiencing. I'd be happy to try implementing such a software on CHIME/FRB's side if that is going to allow more follow-up opportunities.  A related reference detailing the experimental broker can be found here: https://comet.transientskp.org/en/stable/appendix/VOEvent-OpenPGP.html

Best regards, 
-Thomas

**Reply by caseyjlaw** [reply]
I actually implemented a workaround by setting up my own service on an intermediate server. That changes the CHIME "push" architecture into a "pull" from my telescope control server. Not something that everyone can do, but I needed a solution quickly and my admin didn't want to open a port, so I did my workaround.
I'm not sure if others are stuck though.
